src/hooks/Wishlist/wishlist-controller.jsx

```diff
@@ -0,0 +1,46 @@
+export function wishReducerFunction(state, action) {
+
+    switch (action.type) {
+        case "wishlist":
+            let productInWishList = false;
+            let updatedWishList = state.wishItems.map((items) => {
+                if (action.payload.id === items.id) {
+                    productInWishList = true;
+                    return items;
+                }
+                return items;
+            })
+            const { id, title, rating, initialPrice, img, price, discountedPrice } = action.payload
+            if (!productInWishList) {
+                updatedWishList = [...state.wishItems, { id, title, rating, initialPrice, img, price, discountedPrice }]
```

Comment:
i guess here if you simply write action. payload after spreading the state of wishitems as you would have passed payload as the full product ,because you are abstracting the values out of payload in the line 13.also abstraction is also not required here because full product is being added without changing any object key value.  you can try if you have time or else can try from next projects.